File: app/src/main/java/space/pal/sig/old/model/News.java
package space.pal.sig.old.model;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * SpaceNow
 * Created by Catalin on 7/18/2020
 **/
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@Entity(tableName = "news")
public class News {

    @NonNull
    @PrimaryKey
    @ColumnInfo
    private String id;
    @NonNull
    @ColumnInfo
    private String title;
    @NonNull
    @ColumnInfo
    private String description;
    @NonNull
    @ColumnInfo
    private Date date;
    @NonNull
    @ColumnInfo
    private String thumbnail;
    @NonNull
    @ColumnInfo
    private String image;
    @NonNull
    @ColumnInfo
    private String url;
    @NonNull
    @ColumnInfo
    private NewsSource source;

}

File: app/src/main/java/space/pal/sig/model/entity/News.kt
package space.pal.sig.model.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

/**
 * SpaceNow
 * Created by Catalin on 12/24/2020
 **/
@Entity(tableName = "news")
data class News(
        @PrimaryKey(autoGenerate = false)
        @ColumnInfo
        val title: String,
        @ColumnInfo
        val description: String?,
        @ColumnInfo
        val date: Date,
        @ColumnInfo
        val link: String,
        @ColumnInfo
        val image: String?,
        @ColumnInfo
        val thumbnail: String?,
        @ColumnInfo
        val source: NewsSource,
)
